test/unit/org/apache/cassandra/config/ConfigCompatabilityTest.java

```diff
@@ -0,0 +1,369 @@
+import com.google.common.collect.Sets;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.checkerframework.checker.units.qual.K;
+import org.yaml.snakeyaml.introspector.Property;
+
+/**
+ * To create the test files used by this class, run {@link org.apache.cassandra.distributed.upgrade.ConfigCompatabilityTestGenerate}.
+ */
+public class ConfigCompatabilityTest
+{
+    private static final Logger logger = LoggerFactory.getLogger(ConfigCompatabilityTest.class);
+
+    public static final String TEST_DIR = "test/data/config";
+
+    // see CASSANDRA-11115
+    private static final Set<String> THRIFT = ImmutableSet.of("rpc_server_type",
+                                                              "rpc_port",
+                                                              "rpc_listen_backlog",
+                                                              "start_rpc",
+                                                              "thrift_max_message_length_in_mb",
+                                                              "rpc_max_threads",
+                                                              "rpc_min_threads",
+                                                              "rpc_recv_buff_size_in_bytes",
+                                                              "rpc_send_buff_size_in_bytes",
+                                                              "thrift_framed_transport_size_in_mb",
+                                                              "thrift_prepared_statements_cache_size_mb",
+                                                              "request_scheduler",
+                                                              "request_scheduler_id",
+                                                              "request_scheduler_options");
+    // see // CASSANDRA-16956
+    private static final Set<String> WINDOWS = ImmutableSet.of("windows_timer_interval");
+    /**
+     * Breaking Change: These properties were dead code but removed in a minor (4.0.2).  This broke our deprecation rules
+     * as they should have been removed in 5.0 as previously agreed..
```

Comment:
Haven't started really a review, but just wanted to mention - those had to be removed in 4.0 as dead code due to the rewriting of that solution. There is no real deprecation cycle as during it those would have been dead code and not deprecated. Deprecated is when they still work but we plan to remove them. In this particular case we replace one solution with another. Leaving as deprecated parameters will mean to the users "Oh I still can use those" and they will have wrong expectations when actually nothing happens. So in this case this was a bit of dangerous bug/left over

Reply:
🤷 there are others we leave in as removing would be a breaking change... so removing after 4.0.0 is still a breaking change...  this was pointed out in `Re: [RELEASE] Apache Cassandra 4.0.2 released` and see you created https://issues.apache.org/jira/browse/CASSANDRA-17377 to revert the change in 4.0, but seems you didn't add to trunk with this comment

>  I will take care of trunk separately together with docs, etc as we talked in Slack.

so... is this getting reverted so they are back?  or is the plan to remove in 4.1?

Reply:
Handled already